Sources/XCTest/PerformanceMeter.swift

```diff
@@ -0,0 +1,190 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//
+//  PerformanceMeter.swift
+//  Measures the performance of a block of code and reports the results.
+//
+
+#if os(Linux) || os(FreeBSD)
+    import Foundation
+#else
+    import SwiftFoundation
+#endif
+
+/// Describes a type that is capable of measuring some aspect of code performance
+/// over time.
+internal protocol PerformanceMetric {
+    /// Called once per iteration immediately before the tested code is executed. 
+    /// The metric should do whatever work is required to begin a new measurement.
+    func startMeasuring()
+
+    /// Called once per iteration immediately after the tested code is executed.
+    /// The metric should do whatever work is required to finalize measurement.
+    func stopMeasuring()
+
+    /// Called once, after all measurements have been taken, to provide feedback
+    /// about the collected measurements.
+    /// - Returns: Measurement results to present to the user.
+    func calculateResults() -> String
+
+    /// Called once, after all measurements have been taken, to determine whether
+    /// the measurements should be treated as a test failure or not.
+    /// - Returns: A diagnostic message if the results indicate failure, else nil.
+    func failureMessage() -> String?
+}
+
+internal protocol PerformanceMeterDelegate {
+    func recordMeasurements(results: String, file: StaticString, line: UInt)
+    func recordFailure(description: String, file: StaticString, line: UInt)
+    func recordAPIViolation(description: String, file: StaticString, line: UInt)
+}
```

Comment:
The awesome documentation for `PerformanceMetric` spoiled me, now I wish there was documentation here, too 😝